packages/ember-toucan-form/src/-private/types.ts

```diff
@@ -0,0 +1,23 @@
+import type Component from '@glimmer/component';
+import type { ComponentSignatureBlocks } from '@glint/template/-private/signature';
+import type { HeadlessForm } from 'ember-headless-form';
+
+export type UserData = object;
+
+// Glint utilities
+
+export type SignatureOf<C> = C extends typeof Component<infer S> ? S : never;
+
+export type GetBlockParams<
```

Comment:
This helper type is basically the key to infer what our `form` is (yielded from headless-form), by just passing the type of `HeadlessForm` (which is a public type), see L21 below.